polaris/hub/client.py

```diff
@@ -302,14 +308,41 @@ def get_dataset(
         Returns:
             A `Dataset` instance, if it exists.
         """
+        return self._get_dataset(owner, name, ArtifactSubtype.STANDARD.value, verify_checksum)
 
+    def _get_dataset(
+        self,
+        owner: Union[str, HubOwner],
+        name: str,
+        artifact_type: ArtifactSubtype,
+        verify_checksum: bool = True,
+    ) -> Dataset:
+        """Loads either a standard or competition dataset from Polaris Hub
+
+        Args:
+            owner: The owner of the dataset. Can be either a user or organization from the Polaris Hub.
+            name: The name of the dataset.
+            artifact_type: indicates whether the artifact is of the standard or competition type.
+            verify_checksum: Whether to use the checksum to verify the integrity of the dataset.
+
+        Returns:
+            A `Dataset` instance, if it exists.
+        """
         with ProgressIndicator(
-            start_msg="Fetching dataset...",
-            success_msg="Fetched dataset.",
+            start_msg="Fetching artifact...",
+            success_msg="Fetched artifact.",
             error_msg="Failed to fetch dataset.",
         ):
-            response = self._base_request_to_hub(url=f"/dataset/{owner}/{name}", method="GET")
-            storage_response = self.get(response["tableContent"]["url"])
+            url = (
+                f"/v1/dataset/{owner}/{name}"
+                if artifact_type == ArtifactSubtype.STANDARD.value
+                else f"/v2/competition/dataset/{owner}/{name}"
+            )
+            response = self._base_request_to_hub(url=url, method="GET")
+            dataset_info = (
+                "tableContent" if artifact_type == ArtifactSubtype.STANDARD.value else "maskedDatasetInfo"
```

Comment:
**Question**
Could we use the same attribute name in the endpoint response?

Reply:
We could, but we'll need to add the `tableContent` field to the response object just before it is returned from the Hub. I'm currently addressing PR feedback on the Hub so I can add this in.